To do:
- scrape: https://en.wikipedia.org/wiki/Gymnastics_at_the_2020_Summer_Olympics_%E2%80%93_Women%27s_artistic_qualification

- Individual Rankings:
    - https://www.gymnastics.sport/site/rankings/ranking_wag.php
    - https://en.wikipedia-on-ipfs.org/wiki/FIG_World_Rankings

# Top Women:

## Sunisa LEE
- UE 15.300
- BB 13.833

## Jade CAREY
- FX 14.366
- VT 15.200
- VT2 14.300

## Mykayla SKINNER
- VT1 15.033
- VT2 14.800

## Joscelyn ROBERSON
- FX 13.300
- VT1 14.233
- VT2 13.933
- BB 13.233


## Simone BILES
- BB 14.0


## Zoe MILLER
- UB 14.633

## Shilese JONES
- UB 14.766

using data from challenge: 

### Top 5 on BB:
- 2810    ashlee sullivan     USA  13.800
- 2475  joscelyn roberson     USA  13.233
- 1807      nola matthews     USA  13.200
- 1909         zoe miller     USA  12.900


### Top 5 on FX:
- 2469  joscelyn roberson     USA  14.15
- 1808      nola matthews     USA  12.90
- 2811    ashlee sullivan     USA  12.65
- 1910         zoe miller     USA  10.85


### Top 5 on UB:
- 1912         zoe miller     USA  14.85
- 1809      nola matthews     USA  13.65
- 2470  joscelyn roberson     USA  13.00
- 2812    ashlee sullivan     USA  12.55


### Top 5 on VT:
- 2471  joscelyn roberson     USA  14.233
- 1913         zoe miller     USA  13.900
- 1810      nola matthews     USA  13.100
- 2813    ashlee sullivan     USA  13.100




# Men

## Brody MALONE  
- HB 14.800
## Curran PHILLIPS  
- PB 15.500
## Stephen NEDOROSCIK 
- PH 15.233
## Joshua Andrew KARNES 
- VT 14.733
## Asher HONG 
- VT1 15.450

In [2]:
import pandas as pd
import numpy as np

In [2]:
# Zoe MILLER" height unknown
data = {
  "Name": ["Sunisa LEE", "Jade CAREY", "Mykayla SKINNER", "Joscelyn ROBERSON", "Simone BILES", "Zoe MILLER", "Shilese JONES"],
    "DOB": ["03-09-2003", "01-01-2000",  "01-01-1996", "01-01-2006", "03-14-1997", "01-01-2005", "01-01-2002"],
    "Height": [5.0, 5.2, 5.1, 4.8, 4.8, 5.0, 5.3]
}
#load data into a DataFrame object:
df = pd.DataFrame(data)
df

Name         DOB  Height
0         Sunisa LEE  03-09-2003     5.0
1         Jade CAREY  01-01-2000     5.2
2    Mykayla SKINNER  01-01-1996     5.1
3  Joscelyn ROBERSON  01-01-2006     4.8
4       Simone BILES  03-14-1997     4.8
5         Zoe MILLER  01-01-2005     5.0
6      Shilese JONES  01-01-2002     5.3

## Majority of female gymnasts in the modern era in the Olympics are between ages 16 - 22

In [3]:
data = {
  "Name": ["Brody MALONE", "Curran PHILLIPS", "Stephen NEDOROSCIK", "Joshua Andrew KARNES", "Asher HONG"],
    
}

#load data into a DataFrame object:
df1 = pd.DataFrame(data)
df1

Name
0          Brody MALONE
1       Curran PHILLIPS
2    Stephen NEDOROSCIK
3  Joshua Andrew KARNES
4            Asher HONG

Following the competition, USA Gymnastics named the 2023-24 Senior Women’s National Team presented by Xfinity.

Senior National Team Presented by Xfinity
- Simone Biles — Spring, Texas/World Champions Centre
- Skye Blakely — Frisco, Texas/WOGA Gymnastics
- Jordan Chiles — Vancouver, Wash./World Champions Centre
- Kayla DiCello — Boyds, Md./Hill’s Gymnastics
- Amelia Disidore — Overland Park, Kan./Great American Gymnastics Express
- Shi Jones — Auburn, Wash./Ascend Gymnastics Center
- Katelyn Jong — Allen, Texas/Metroplex Gymnastics
- Joscelyn Roberson — Texarkana, Texas/World Champions Centre
- Tiana Sumanasekera — Pleasanton, Calif./World Champions Centre
- Leanne Wong — Overland Park, Kan./University of Florida Gymnastics

https://usagym.org/biles-becomes-all-time-all-around-champion-at-xfinity-u-s-gymnastics-championships/

Current roster
As of August 28, 2023: 
    
    
- Simone Biles 	March 14, 1997 (age 26) 
- Skye Blakely 	February 4, 2005 (age 18)
- Jordan Chiles 	April 15, 2001 (age 22)
- Kayla DiCello 	January 25, 2004 (age 19)
- Amelia Disidore 	September 19, 2006 (16)
- Shilese Jones 	July 26, 2002 (age 21) 
- Katelyn Jong 	July 28, 2006 (age 17) 
- Joscelyn Roberson 	February 8, 2006 (age 17) 
- Tiana Sumanasekera 	September 15, 2007 (age 15) 
- Leanne Wong 	September 20, 2003 (age 19) 


https://en.wikipedia.org/wiki/United_States_women%27s_national_artistic_gymnastics_team

<b>insidegymnastics</b>

On the women’s side, <b>Simone Biles, Suni Lee, Jordan Chiles and Grace McCallum </b>were named to the team, while <b>MyKayla Skinner and Jade Carey</b> were named as individuals. <b>Chiles, Carey and Lee</b> have all announced their plans to train for Tokyo and while there is no official word from Biles yet (her rep tells us she is still making her decision), Biles has been spotted in the gym in photos and videos posted by teammates.

On the men’s side, <b>Sam Mikulak, Brody Malone, Yul Moldauer and Shane Wiskus</b> were named to the team, while <b>Alec Yoder</b> was named as an individual.<b> Mikulak and Yoder have since retired but Malone, Moldauer and Wiskus</b> all have their sights set on Paris 2024.

https://www.insidegymnastics.com/news-features/2024-u-s-olympic-trials-coming-to-twin-cities/

https://www.insidegymnastics.com/news-features/history-in-the-making-douglas-biles-and-lee-set-sights-on-paris-2024/

In [4]:
data = {
  "Name": ["Simone Biles", "Skye Blakely", "Jordan Chiles", "Kayla DiCello", "Amelia Disidore",
          "Shi Jones", "Katelyn Jong", "Joscelyn Roberson", "Tiana Sumanasekera",
          "Leanne Wong", "Suni Lee", "Grace McCallum", "MyKayla Skinner", "Jade Carey"]
 
}
#load data into a DataFrame object:
women = pd.DataFrame(data)
women

Name
0         Simone Biles
1         Skye Blakely
2        Jordan Chiles
3        Kayla DiCello
4      Amelia Disidore
5            Shi Jones
6         Katelyn Jong
7    Joscelyn Roberson
8   Tiana Sumanasekera
9          Leanne Wong
10            Suni Lee
11      Grace McCallum
12     MyKayla Skinner
13          Jade Carey

Team events will feature 12 teams of 5 women

Composition Requirements (CR): In the 2017 Code of Points, gymnasts must demonstrate skills from four required Element Groups on each apparatus. For each composition requirement completed, 0.5 points are awarded. The maximum CR score is 2.00. From 2008 to 2016, there were five different Element Groups for each apparatus, allowing a gymnast to earn up to 2.5 points.

https://en.wikipedia.org/wiki/Code_of_Points_(gymnastics)

In [5]:
data = {
    "us_champiuonship2023women_day1results": ["simone biles",
                                              "shilese jones",
                                              "skye blakely",
                                              "leanne wong",
                                              "jordan chiles",
                                              "kaliya lincoln",
                                              "Amelia disidore",
                                              "Joscelyn Roberson",
                                              "katelyn jong",
                                              "Madray Johnson",
                                              "tiana sumanasekera"]
}

#load data into a DataFrame object:
day1 = pd.DataFrame(data)
day1

us_champiuonship2023women_day1results
0                           simone biles
1                          shilese jones
2                           skye blakely
3                            leanne wong
4                          jordan chiles
5                         kaliya lincoln
6                        Amelia disidore
7                      Joscelyn Roberson
8                           katelyn jong
9                         Madray Johnson
10                    tiana sumanasekera

### 2023 us champiuonship - women day 1 results august 25, 2023

1. simone biles
2. shilese jones
3. skye blakely
4. leanne wong
5. jordan chiles
6. kaliya lincoln
7. Amelia disidore
8. Joscelyn Roberson
9. katelyn jong
10. Madray Johnson
11. tiana sumanasekera
---
19. tie Kayla DiCello
---
28. Sunisa Lee

In [6]:
data = {
    "us_champiuonship2023women_day2results": ["simone biles",
                                              "shilese jones",
                                              "leanne wong",
                                              "skye blakely",
                                              "jordan chiles",
                                              "katelyn jong",
                                              "Joscelyn Roberson",
                                              "tiana sumanasekera",
                                              "kayla dicello",
                                              "Amelia disidore",
                                              "kaliya lincoln",
                                              "Madray Johnson"
                                              ]
}

#load data into a DataFrame object:
day2 = pd.DataFrame(data)
day2

us_champiuonship2023women_day2results
0                           simone biles
1                          shilese jones
2                            leanne wong
3                           skye blakely
4                          jordan chiles
5                           katelyn jong
6                      Joscelyn Roberson
7                     tiana sumanasekera
8                          kayla dicello
9                        Amelia disidore
10                        kaliya lincoln
11                        Madray Johnson

### 2023 us champiuonship - women day 2 results august 27, 2023

1. simone biles
2. shilese jones
3. leanne wong
4. skye blakely
5. jordan chiles
6. kately jong
7. Joscelyn Roberson
8. tie tiana sumanasekera
8. tie kayla dicello
10. Amelia disidore
---
27. tie sunisa lee

In [7]:
# Concatenate the DataFrames along columns (axis=1)
combined_df = pd.concat([women, day1, day2], axis=1)

# Reset the index of the combined DataFrame
combined_df.reset_index(drop=True, inplace=True)

df = combined_df

In [8]:
# df['name_nums'] = range(1, 15)
# df['day1results_nums'] = range(1, 15)
# df['day2results_nums'] = range(1, 15)

In [9]:
# df

In [10]:
# df.columns

In [11]:
# new_order = ['Name',  'name_nums','us_champiuonship2023women_day1results',
#        'day1results_nums', 'us_champiuonship2023women_day2results',
#         'day2results_nums']

# # Create a new DataFrame with the columns in the new order
# df = df[new_order]
# df

In [12]:
# df = df.drop(columns=["Name", "name_nums"])
# df

In [13]:
# df["Name"] = df["us_champiuonship2023women_day2results"]
# df = df.drop(columns=["us_champiuonship2023women_day2results"])

# df

## average placement from day 1 and 2 of usa championships, and of random person on internets ranking

In [24]:
# # Create a dictionary to store the sum of index values for each person
# index_sum = {}

# # Iterate through each row and column to calculate the sum of index values
# for index, row in df.iterrows():
#     name = row['Name']
#     if name not in index_sum:
#         index_sum[name] = 0
#     index_sum[name] += index

# # Calculate the average index value for each person
# average_index = {name: index_sum[name] / df[df['Name'] == name].shape[0] for name in index_sum}

# # Create a DataFrame from the dictionary
# average_index_df = pd.DataFrame(average_index.items(), columns=['Name', 'Average_Index'])

# # Display the average index values
# print(average_index_df)



In [16]:
df = df.drop(columns=["Name"])
df.head()

us_champiuonship2023women_day1results us_champiuonship2023women_day2results
0                          simone biles                          simone biles
1                         shilese jones                         shilese jones
2                          skye blakely                           leanne wong
3                           leanne wong                          skye blakely
4                         jordan chiles                         jordan chiles

In [17]:
df["Name"] = df["us_champiuonship2023women_day2results"]
df.head()

us_champiuonship2023women_day1results us_champiuonship2023women_day2results  \
0                          simone biles                          simone biles   
1                         shilese jones                         shilese jones   
2                          skye blakely                           leanne wong   
3                           leanne wong                          skye blakely   
4                         jordan chiles                         jordan chiles   

            Name  
0   simone biles  
1  shilese jones  
2    leanne wong  
3   skye blakely  
4  jordan chiles

In [18]:
df = df.drop(columns=["us_champiuonship2023women_day2results"])
df

us_champiuonship2023women_day1results                Name
0                           simone biles        simone biles
1                          shilese jones       shilese jones
2                           skye blakely         leanne wong
3                            leanne wong        skye blakely
4                          jordan chiles       jordan chiles
5                         kaliya lincoln        katelyn jong
6                        Amelia disidore   Joscelyn Roberson
7                      Joscelyn Roberson  tiana sumanasekera
8                           katelyn jong       kayla dicello
9                         Madray Johnson     Amelia disidore
10                    tiana sumanasekera      kaliya lincoln
11                                   NaN      Madray Johnson
12                                   NaN                 NaN
13                                   NaN                 NaN

In [19]:
new_order = ['Name', 'us_champiuonship2023women_day1results']

# Create a new DataFrame with the columns in the new order
df = df[new_order]
df

Name us_champiuonship2023women_day1results
0         simone biles                          simone biles
1        shilese jones                         shilese jones
2          leanne wong                          skye blakely
3         skye blakely                           leanne wong
4        jordan chiles                         jordan chiles
5         katelyn jong                        kaliya lincoln
6    Joscelyn Roberson                       Amelia disidore
7   tiana sumanasekera                     Joscelyn Roberson
8        kayla dicello                          katelyn jong
9      Amelia disidore                        Madray Johnson
10      kaliya lincoln                    tiana sumanasekera
11      Madray Johnson                                   NaN
12                 NaN                                   NaN
13                 NaN                                   NaN

In [23]:
df = df.iloc[:12]
df

Name us_champiuonship2023women_day1results
0         simone biles                          simone biles
1        shilese jones                         shilese jones
2          leanne wong                          skye blakely
3         skye blakely                           leanne wong
4        jordan chiles                         jordan chiles
5         katelyn jong                        kaliya lincoln
6    Joscelyn Roberson                       Amelia disidore
7   tiana sumanasekera                     Joscelyn Roberson
8        kayla dicello                          katelyn jong
9      Amelia disidore                        Madray Johnson
10      kaliya lincoln                    tiana sumanasekera
11      Madray Johnson                                   NaN

In [25]:
df

Name us_champiuonship2023women_day1results
0         simone biles                          simone biles
1        shilese jones                         shilese jones
2          leanne wong                          skye blakely
3         skye blakely                           leanne wong
4        jordan chiles                         jordan chiles
5         katelyn jong                        kaliya lincoln
6    Joscelyn Roberson                       Amelia disidore
7   tiana sumanasekera                     Joscelyn Roberson
8        kayla dicello                          katelyn jong
9      Amelia disidore                        Madray Johnson
10      kaliya lincoln                    tiana sumanasekera
11      Madray Johnson                                   NaN

In [26]:
# df.to_csv('us_champ_2023.csv', index=False)

In [22]:
# # Create a dictionary to store the sum of index values for each person
# index_sum = {}

# # Create a dictionary to count the occurrences of each person
# person_count = {}

# # Iterate through each row and column (except the 'Name' column) to calculate the sum of index values
# for index, row in df.iloc[:, 1:].iterrows():
#     name = df.at[index, 'Name']  # Get the 'Name' value for the current row
#     if pd.notna(name):  # Exclude rows with NaN in the 'Name' column
#         if name not in index_sum:
#             index_sum[name] = 0
#             person_count[name] = 0
#         index_sum[name] += index
#         person_count[name] += 1

# # Calculate the average index value for each person
# average_index = {name: round(index_sum[name] / person_count[name], 2) for name in index_sum}

# # Create a DataFrame from the dictionary
# average_index_df = pd.DataFrame(average_index.items(), columns=['Name', 'Average_Index'])

# # Display the average index values rounded to two decimal places
# print(average_index_df)



## cleaned in excel

In [27]:
df = pd.read_csv('us_champ_2023.csv')
df

Name  day2rank day1rank
0         simone biles         1        1
1        shilese jones         2        2
2          leanne wong         3        4
3         skye blakely         4        3
4        jordan chiles         5        5
5         katelyn jong         6        9
6    Joscelyn Roberson         7        8
7   tiana sumanasekera         8       11
8        kayla dicello         9        -
9      Amelia disidore        10        7
10      kaliya lincoln        11        6
11      Madray Johnson        12       10

In [32]:
# Fill NaN values in the DataFrame with 15
df.replace('-', 15, inplace=True)
df

Name  day2rank day1rank
0         simone biles         1        1
1        shilese jones         2        2
2          leanne wong         3        4
3         skye blakely         4        3
4        jordan chiles         5        5
5         katelyn jong         6        9
6    Joscelyn Roberson         7        8
7   tiana sumanasekera         8       11
8        kayla dicello         9       15
9      Amelia disidore        10        7
10      kaliya lincoln        11        6
11      Madray Johnson        12       10

In [33]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 12 entries, 0 to 11
Data columns (total 3 columns):
 #   Column    Non-Null Count  Dtype 
---  ------    --------------  ----- 
 0   Name      12 non-null     object
 1   day2rank  12 non-null     int64 
 2   day1rank  12 non-null     object
dtypes: int64(1), object(2)
memory usage: 416.0+ bytes


In [34]:
df['day1rank'] = df['day1rank'].astype(int)

In [35]:
# Calculate the average of 'day2rank' and 'day1rank' columns
df['average_rank'] = (df['day2rank'] + df['day1rank']) / 2

df

Name  day2rank  day1rank  average_rank
0         simone biles         1         1           1.0
1        shilese jones         2         2           2.0
2          leanne wong         3         4           3.5
3         skye blakely         4         3           3.5
4        jordan chiles         5         5           5.0
5         katelyn jong         6         9           7.5
6    Joscelyn Roberson         7         8           7.5
7   tiana sumanasekera         8        11           9.5
8        kayla dicello         9        15          12.0
9      Amelia disidore        10         7           8.5
10      kaliya lincoln        11         6           8.5
11      Madray Johnson        12        10          11.0

## Top 5 
1. simone biles 1.0
2. shilese jones 2.0
3. leanne wong 3.5
4. skye blakely 3.5
5. jordan chiles 5.0

0 	Simone Biles
1 	Skye Blakely
2 	Jordan Chiles

3 	Kayla DiCello
4 	Amelia Disidore
5 	Shi Jones
6 	Katelyn Jong
7 	Joscelyn Roberson
8 	Tiana Sumanasekera
9 	Leanne Wong

Artistic Gymnastics World Championships - Antwerp (BEL)


Team women

    ROBERSON Joscelyn - Team
    CAREY Jade - Team
    DI CELLO Kayla - Team
    SUMANASEKERA Tiana - Team
    MILLER Zoe - Team
    MATTHEWS Nola - Team

https://antwerpgymnastics2023.com/en/event/athletes-teams/united-states-america


## 2020 olympic team
2nd place, silver medalist(s)

- United States 	42.732 (4) 	43.266 (2) 	41.232 (2) 	38.866 (8) 	166.096
    - Simone Biles (USA) 	13.766 			
    - Jordan Chiles (USA) 	14.666 	14.166 	13.433 	11.700
    - Sunisa Lee (USA) 		15.400 	14.133 	13.666
    - Grace McCallum (USA) 	14.300 	13.700 	13.666 	13.500 
    
## 2016 team
Gymnast 	Vault 	Bars 	Beam 	Floor 	Total

- United States 	46.866 (1) 	46.499 (1) 	45.533 (1) 	45.999 (1) 	184.897 (1) 
    - Simone Biles 	15.933 	14.800 	15.300 	15.800 	61.833
    - Gabby Douglas 		15.766 			15.766
    - Laurie Hernandez 	15.100 		15.233 	14.833 	45.166
    - Madison Kocian 		15.933 			15.933
    - Aly Raisman 	15.833 		15.000 	15.366 	46.199


## 2012 team
1st place, gold medalist(s) 	 

- United States 	48.132 (1) 	44.799 (3) 	45.299 (1) 	45.366 (1) 	183.596
    - Gabby Douglas 	15.966 	15.200 	15.233 	15.066
    - Jordyn Wieber 	15.933 	14.666 		15.000
    - Aly Raisman 			14.933 	15.300
    - Kyla Ross 		14.933 	15.133 	
    - McKayla Maroney 	16.233 		
    
    
## 2008 team

- United States 	46.875 (1) 	47.975 (2) 	47.250 (1) 	44.425 (3) 	186.525
    - Shawn Johnson 	16.000 	15.350 	16.175 	15.100
    - Nastia Liukin 		16.900 	15.975 	15.200
    - Chellsie Memmel 		15.725 		
    - Samantha Peszek 				
    - Alicia Sacramone 	15.675 		15.100 	14.125
    - Bridget Sloan 	15.200 			
    
    
## 2004
- United States (USA) 	28.387 (2) 	28.524 (1) 	28.499 (3) 	28.174 (3) 	113.584
    - Mohini Bhardwaj 	9.500 		9.400 	9.325 	28.225
    - Annia Hatch 	9.562 				9.562
    - Terin Humphrey 		9.575 	9.487 		19.062
    - Courtney Kupets 		9.662 		9.187 	18.849
    - Courtney McCool 					
    - Carly Patterson 	9.325 	9.287 	9.612 	9.662 	37.886 
    
    
## 2000
- United States (USA) 	37.685 (4) 	38.875 (1) 	37.699 (5) 	38.674 (3) 	152.933
    - Amy Chow 	9.350 	9.750 	9.112 	9.625 	37.837
    - Elise Ray 	9.418 	9.725 	9.337 	9.337 	37.817
    - Jamie Dantzscher 	9.424 	9.700 		9.712 	28.836
    - Kristen Maloney 	9.493 		9.537 	9.737 	28.767
    - Tasha Schwikert 		9.675 	9.350 	9.600 	28.625
    - Dominique Dawes 	9.243 	9.700 	9.475 		28.418 
    
## 1996
- United States (USA) 	97.209 	2 	97.809 	1 	96.410 	2 	97.797 	1 	389.225
    - Shannon Miller 	9.762 	9.700 	10 	9.775 	9.787 	9 	9.737 	9.862 	1 	9.787 	9.618 	18 	78.028 	2
    - Dominique Dawes 	9.725 	9.762 	8 	9.762 	9.850 	5 	9.425 	9.725 	11 	9.687 	9.850 	7 	77.786 	6
    - Kerri Strug* 	9.812 	9.712 	5 	9.675 	9.787 	14 	9.350 	9.737 	15 	9.825 	9.837 	1 	77.735 	7
    - Dominique Moceanu 	9.662 	9.200 	48 	9.725 	9.812 	11 	9.687 	9.850 	2 	9.750 	9.837 	4 	77.523 	11
    - Jaycie Phelps 	9.587 	9.662 	21 	9.712 	9.787 	13 	9.012 	9.600 	38 	9.662 	9.750 	15 	76.722 	17
    - Amy Chow 	9.700 	9.712 	11 	9.762 	9.837 	6 	  	  	39.011 	95
    - Amanda Borden 	  	  	9.312 	9.725 	19 	9.712 	9.762 	10 	38.511 	97 

In [5]:
data = {
    "oly2020": ["simone biles","jordan chiles", "Sunisa Lee", "Grace McCallum"],
    "oly2016": ["simone biles", "Gabby Douglas", "Laurie Hernandez", "Madison Kocian", "Aly Raisman"],
    "oly2012": ["Gabby Douglas", "Jordyn Wieber", "Aly Raisman", "Kyla Ross", "McKayla Maroney"],
    "oly2008": ["Shawn Johnson", "Nastia Liukin", "Chellsie Memmel", "Samantha Peszek", "Alicia Sacramone", "Bridget Sloan"],
    "oly2004": ["Mohini Bhardwaj", "Annia Hatch", "Terin Humphrey", "Courtney Kupets", "Courtney McCool", "Carly Patterson"],
    "oly2000": ["Amy Chow", "Elise Ray", "Jamie Dantzsche", "Kristen Maloney", "Tasha Schwikert", "Dominique Dawes"],
    "oly1996": ["Shannon Miller", "Dominique Dawes", "Kerri Strug", "Dominique Moceanu", "Jaycie Phelps", "Amy Chow", "Amanda Borden"]
}

# Find the maximum length of the lists
max_length = max(len(lst) for lst in data.values())

# Fill the lists with None to make them of equal length
for key in data:
    data[key] += [None] * (max_length - len(data[key]))

df = pd.DataFrame(data)

df


oly2020           oly2016          oly2012           oly2008  \
0    simone biles      simone biles    Gabby Douglas     Shawn Johnson   
1   jordan chiles     Gabby Douglas    Jordyn Wieber     Nastia Liukin   
2      Sunisa Lee  Laurie Hernandez      Aly Raisman   Chellsie Memmel   
3  Grace McCallum    Madison Kocian        Kyla Ross   Samantha Peszek   
4            None       Aly Raisman  McKayla Maroney  Alicia Sacramone   
5            None              None             None     Bridget Sloan   
6            None              None             None              None   

           oly2004          oly2000            oly1996  
0  Mohini Bhardwaj         Amy Chow     Shannon Miller  
1      Annia Hatch        Elise Ray    Dominique Dawes  
2   Terin Humphrey  Jamie Dantzsche        Kerri Strug  
3  Courtney Kupets  Kristen Maloney  Dominique Moceanu  
4  Courtney McCool  Tasha Schwikert      Jaycie Phelps  
5  Carly Patterson  Dominique Dawes           Amy Chow  
6             None             None      Amanda Borden